subsys/llext/llext_link.c

```diff
@@ -172,44 +179,72 @@ int llext_link(struct llext_loader *ldr, struct llext *ext, bool do_local)
 	for (i = 0; i < ldr->sect_cnt; ++i) {
 		elf_shdr_t *shdr = ldr->sect_hdrs + i;
 
-		/* find relocation sections */
-		if (shdr->sh_type != SHT_REL && shdr->sh_type != SHT_RELA) {
+		/* find proper relocation sections */
+		switch (shdr->sh_type) {
+		case SHT_REL:
+			if (shdr->sh_entsize != sizeof(elf_rel_t)) {
+				LOG_ERR("Invalid relocation entry size %zd for section %d",
+					(size_t)shdr->sh_entsize, i);
+				return -EINVAL;
+			}
+			break;
+		case SHT_RELA:
```

Comment:
Nit: Considering we don't actually seem to handle `.rela` sections properly (`elf_rela_t.r_addend` appears unused in the whole codebase?), maybe this whole branch should return `-ENOTSUP` for now.

Reply:
I would love to, but I would surely break something in the Xtensa port - it actually has code dealing with them (and ignoring `r_addend`) just below:
https://github.com/zephyrproject-rtos/zephyr/blob/a869972e0ca070d6686614438292dbd7703b14d5/subsys/llext/llext_link.c#L226-L227